### PR TITLE
perf: improve the initial page load speed

### PR DESCRIPTION
**Pull request by keita-determined**
## Ticket

[ET-783]

## Description

Initial page load speed is slow in any pages, especially the flat run table page. In order to fix this issue, applied 2 techniques. 
1. Code splitting
2. Compress CSS, JS files (surprisingly these files weren't compressed)

### Google Chrome LightHouse

Backend is http://latest-main.determined.ai:8080/
Tested web page is http://localhost:8080/det/projects/1/runs

#### Before
Screenshot 2024-09-16 at 1 19 37 PM

#### After
Screenshot 2024-09-16 at 1 59 40 PM

## Test Plan

- Check if JS/CSS files are compressed (gzip)
- Check if web files are splitting
- Check if all pages are accessible like before (no regressions)

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-783]: https://hpe-aiatscale.atlassian.net/browse/ET-783?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by corban-beaird — APPROVED** [review]
Looks solid!

**Comment by ashtonG** [comment]
having trouble reproducing your lighthouse results -- here's what i get running your code locally:
image

could you specify the lighthouse settings?

**Comment by keita-determined** [comment]
> having trouble reproducing your lighthouse results -- here's what i get running your code locally: image
> 
> could you specify the lighthouse settings?

@ashtonG 
Lighthouse needs to be run on localhost:8080 after `make build` in web/react folder.

File: master/internal/core.go [hunk 1]
```diff
@@ -1284,7 +1284,7 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 	m.echo.Use(middleware.Recover())
 
 	// Files that receive a unique hash when bundled and deployed can be cached forever
-	cacheFileLongTerm := regexp.MustCompile(`(-[0-9a-z]{1,}\.(js|css))$|(woff2|woff)$`)
+	cacheFileLongTerm := regexp.MustCompile(`(-[0-9a-zA-Z]{1,}\.(js|css))$|(woff2|woff)$`)
```
[the remark this answers is not in the document]
Comment (keita-determined) [reply master/internal/core.go]: make sense. fixed

**Comment by ashtonG** [comment]
I'm sorry, i'm still not seeing it. can we chat over slack to troubleshoot?
image

**Pull request merged**